lib/local-constructs/lambda/index.ts

```diff
@@ -0,0 +1,186 @@
+import { Construct } from "constructs";
+import {
+  NodejsFunction,
+  NodejsFunctionProps,
+} from "aws-cdk-lib/aws-lambda-nodejs";
+import { Duration, RemovalPolicy, Stack, CfnElement } from "aws-cdk-lib";
+import { LogGroup } from "aws-cdk-lib/aws-logs";
+import { Runtime } from "aws-cdk-lib/aws-lambda";
+import {
+  Effect,
+  ManagedPolicy,
+  PolicyDocument,
+  PolicyStatement,
+  Role,
+  ServicePrincipal,
+} from "aws-cdk-lib/aws-iam";
+import { commonBundlingOptions } from "../../config/bundling-config";
+import * as apigateway from "aws-cdk-lib/aws-apigateway";
+import * as dynamodb from "aws-cdk-lib/aws-dynamodb";
+
+interface LambdaProps extends Partial<NodejsFunctionProps> {
+  handler?: string;
+  timeout?: Duration;
+  memorySize?: number;
+  brokerString?: string;
+  path?: string;
+  method?: string;
+  stackName: string;
+  tables: { [name: string]: dynamodb.Table };
+  api: apigateway.RestApi;
+}
+
+export class Lambda extends Construct {
+  public readonly lambda: NodejsFunction;
+
+  constructor(scope: Construct, id: string, props: LambdaProps) {
+    super(scope, id);
+
+    const {
+      handler = "main",
+      timeout = Duration.seconds(6),
+      memorySize = 1024,
+      brokerString = "",
+      path,
+      method,
+      ...restProps
+    } = props;
+
+    const logGroup = new LogGroup(this, `${id}LogGroup`, {
+      removalPolicy: RemovalPolicy.DESTROY,
+    });
+
+    const stage = this.node.tryGetContext("stage") || "dev";
+
+    const environment = {
+      BOOTSTRAP_BROKER_STRING_TLS: brokerString,
+      STAGE: stage,
+      stage,
+      ...Object.values(props.tables).reduce((acc, table) => {
+        const currentTable = Stack.of(table)
+          .getLogicalId(table.node.defaultChild as CfnElement)
+          .slice(0, -8);
+
+        acc[`${currentTable}Name`] = table.tableName;
+        acc[`${currentTable}Arn`] = table.tableArn;
+
+        return acc;
+      }, {} as { [key: string]: string }),
+    };
```

Comment:
Are there any env vars for lambdas that aren't tables? I'll have to look but I would guess there's at least one. Maybe the pdf gen one.

Reply:
There's the broker string and stage.